spec/keyrod/oidc_client_spec.rb

```diff
@@ -0,0 +1,24 @@
+require 'spec_helper'
+
+describe Keyrod::OIDCClient do
+  subject(:oidc_client) { described_class.new }
+
+  before do
+    Keyrod::Settings['oidc-site'] = 'https://aai-dev.egi.eu/oidc/token'
+    Keyrod::Settings['refresh-token'] = 'iOiI0ZDQ5MzUxMC00MDJhLTQ3NDQtYjE4Yi'
+    Keyrod::Settings['client-id'] = 'keyrod'
+    Keyrod::Settings['client-secret'] = '3HcNV88ex48UKYPWCI1OwresZAxMjh1Pdf'
+  end
+
+  describe '#new' do
```

Comment:
Maybe you should also test whether those object variables are really set as you expect (judging from the constructor implementation).